File: tests/fixtures/openai-realtime/openai-realtime-migrate-beta-to-ga-broken/01-caller-socket.ts
import WebSocket from 'ws';

function setupCallerSocket(apiKey: string): WebSocket {
  const url = 'wss://api.openai.com/v1/realtime?model=gpt-4o-realtime-preview-2024-10-01';
  const callerSocket = new WebSocket(url, {
    headers: {
      Authorization: `Bearer ${apiKey}`,
      'OpenAI-Beta': 'realtime=v1',
    },
  });
  return callerSocket;
}

File: tests/fixtures/openai-realtime/openai-realtime-migrate-beta-to-ga-broken/01_caller_socket.py
import asyncio
import os

import websockets

WS_URL = "wss://api.openai.com/v1/realtime?model=gpt-4o-realtime-preview"
HEADERS = {
    "Authorization": f"Bearer {os.environ.get('OPENAI_API_KEY')}",
    "OpenAI-Beta": "realtime=v1",
}


async def connect_caller_socket():
    async with websockets.connect(WS_URL, additional_headers=HEADERS) as ws:
        async for message in ws:
            print(message)


asyncio.run(connect_caller_socket())
